### ISSUE TITLE
Perception warnings in State.init 

### ISSUE THREAD

**Issue by KirillZholnerovich** [pointfreeco/swift-composable-architecture#3402]
### Description

For example, we have ParentFeature and ChildFeature and if we have array of ChileFeature.State in ParentFeature.State it throws runtime perception warning

Code:
 ```
@Reducer
 public struct ParentFeature {
	@ObservableState
	public struct State {
		public var items: IdentifiedArrayOf<ChildFeature.State> = [ .init() ]
		
		//...
	}

    //.......
 }
```

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

No perception warnings

### Actual behavior

Have perception warnings

### The Composable Architecture version information

1.15.0

### Destination operating system

iOS 18.0

### Xcode version information

Xcode 16.0

### Swift Compiler version information

```shell
swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0
```

**Comment by stephencelis** [comment]
@KirillZholnerovich I don't think this is a bug in the library and you have not provided a full reproduction for us to evaluate the problem.

It does sound like maybe there could be an observation issue in your application requiring a `WithPerceptionTracking` view, but we'll need more info to help. Please follow up in the discussion with some compiling sample code!
